### ISSUE TITLE
Package DiffFit as a standard Python wheel

### ISSUE THREAD

**Issue by RodenLuo** [nanovis/DiffFitViewer#11]
https://www.cgl.ucsf.edu/chimerax/docs/devel/writing_bundles.html

> With ChimeraX bundles being packages as standard Python wheel-format files, they can be distributed as plain files and installed using the ChimeraX `toolshed install` command. Thus, electronic mail, web sites and file sharing services can all be used to distribute ChimeraX bundles.

**Comment by RodenLuo** [comment]
@tomgoddard

I can install the wheel via the following command: 

```cmd
ChimeraX\bin>ChimeraX-console.exe --nogui --cmd "toolshed install D:\GIT\DiffFitViewer\dist\DiffFitViewerTool-0.1-py3-none-any.whl"
Replacing fetcher for 'pdb_nmr' and format NMRSTAR from NMRSTAR bundle with that from NMRSTAR bundle
Executing: toolshed install D:\GIT\DiffFitViewer\dist\DiffFitViewerTool-0.1-py3-none-any.whl
bundle 'D:\\GIT\\DiffFitViewer\\dist\\DiffFitViewerTool-0.1-py3-none-any.whl' already installed
UCSF ChimeraX version: 1.7.1 (2024-01-23)
© 2016-2023 Regents of the University of California.  All rights reserved.
```

I then see the bundle under `Tools > Volume Data > DiffFit`. When I try to launch it, it complains about the missing Python packages. 

```
chimerax.core.toolshed.ToolshedError: start_tool() failed for tool DiffFit in bundle DiffFitViewerTool:
No module named 'sklearn'
```

DiffFit needs `PyTorch, Biopython, mrcfile, and scikit-learn`. It can be installed via the following commands. One might need to change the command depending on the availability of CUDA and its version for PyTorch. 

```cmd
ChimeraX\bin\python.exe -m pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu121

ChimeraX\bin\python.exe -m pip install biopython mrcfile scikit-learn
```

How should I proceed so that it will make the user experience smoother? 

How can I uninstall the bundle installed from a file? I tried with the wheel path or the name `DiffFit`, but it is not working. 

```cmd
ChimeraX\bin>ChimeraX-console.exe --nogui --cmd "toolshed uninstall D:\GIT\DiffFitViewer\dist\DiffFitViewerTool-0.1-py3-none-any.whl ; exit"
...
chimerax.core.errors.UserError: Missing or invalid "bundleNames" argument: Invalid bundle name

ChimeraX\bin>ChimeraX-console.exe --nogui --cmd "toolshed uninstall DiffFit; exit"
Executing: toolshed uninstall DiffFit
"DiffFit" does not match any bundles
```

How can I uninstall a bundle installed via `devel build D:\GIT\DiffFitViewer; devel install D:\GIT\DiffFitViewer`? 

Thanks

**Comment by RodenLuo** [comment]
Hi Tom, Many thanks! I now generated a wheel and released it at this repo's release page. 

TL;DR:

I saw the dependencies section before, but I thought they were other ChimeraX bundles. Later, I was not sure how to add the specific `--index-url https://download.pytorch.org/whl/cu121` for `pip` to have the CUDA version of PyTroch. It seems that in the existence of GPU, `pip` picks up the CUDA version automatically. 

Regarding the uninstall, I was puzzled by the _bundle-name_ as it was first mentioned it could be "a file name" in the doc. I then tried the `toolshed uninstall` command with the file name as shown before. Since it was not shown in the same italic type setting under `toolshed list`, and somehow the listed result cannot be used in `toolshed show`, I did not link them altogether and thought _bundle-name_ was either a file name or something specified on the toolshed website. 

![image](https://github.com/nanovis/DiffFitViewer/assets/8460424/05281506-5d89-4edc-9844-83b0d36fc929)

I understand everybody is very busy. Hopefully, you can see that I'm not trying to exploit you as a ChimeraX expert without reading the doc by myself first, but more hoping to have you in a catalyst-like role, in the sense that your 1-5 minutes could lower the barriers, sometimes causing hours to days of searching and trying.

**Comment by tomgoddard** [comment]
Hi Roden,

  The "toolshed show" command is documented to "show the bundle in the toolshed" as you can see by using the "usage toolshed show" command:

> **usage toolshed show**
> **toolshed show** [bundleName]
>     — show the toolshed or bundle in toolshed
>   bundleName: a text string

In other words it shows the Toolshed page in the ChimeraX web browser for the named bundle.  AddCharge and the other ChimeraX built-in bundles distributed with ChimeraX are not in the Toolshed and that is why you get the message:
"Cannot find bundle AddCharge in Toolshed".

  I agree it is a bit weird that "toolshed list" lists these built-in bundles that are not actually in the Toolshed.

  Documentation for the toolshed command is here

    https://www.cgl.ucsf.edu/chimerax/docs/user/commands/toolshed.html

  I'm not clear if your previous comment is asking any questions.  I tried installing your DiffFIt 0.1 wheel which worked and uninstalling it with "toolshed uninstall DiffFit" worked.

  Don't take my previous rant about not asking so many questions too seriously.  It is fine to ask questions that save you a day of time when it will only take 5 minutes.  But there is a lot of detailed ChimeraX documentation online, and searching that and reading the relevant parts before asking questions is what I was suggesting.

   Tom
